### DISCUSSION TITLE
Need help in certificate installation in IDS

### DISCUSSION THREAD

**Discussion by iamynr5** [International-Data-Spaces-Association/DataspaceConnector#91] [General]
HI Guys,

I have downloaded the certificates[.p12/.cert/.key] given by fraunhour & placed it in conf folders
As per your documentation i have changed the below in config.json & appilication.properties file, 
After changing i have rebuild it again with ./maven clean package, build failure occurs with the below error message
can you any guide me?

**Config.json**

"ids:connectorDeployMode" : {
"@id" : "idsc:**PRODUCTIVE_DEPLOYMENT**"
},
"ids:connectorDescription" : {
"@type" : "ids:BaseConnector",
"@id" : "**https://test.ids.com**",

"ids:hasDefaultEndpoint" : {
"@type" : "ids:ConnectorEndpoint",
"@id" : "https://w3id.org/idsa/autogen/connectorEndpoint/e5e2ab04-633a-44b9-87d9-a097ae6da3cf",
"ids:accessURL" : {
"@id" : "**https://test.ids.com/api/ids/data**"
}
"ids:connectorStatus" : {
"@id" : "idsc:**CONNECTOR_ONLINE**"
},
"ids:keyStore" : {
"@id" : "file:///conf/**test.ids.com.p12**"
}
**appilication.properties**
TLS

server.ssl.enabled=true
server.ssl.key-store-type=PKCS12
server.ssl.key-store=classpath:conf/**test.ids.com.p12**
server.ssl.key-store-password=password
server.ssl.key-alias=1
#security.require-ssl=true
IDSCP

application.http.base-url=https://**test.ids.com**
idscp2.enabled=false
idscp2.keystore=./src/main/resources/conf/**test.ids.com.p12**
idscp2.truststore=./src/main/resources/conf/truststore.p12
**Error**

[INFO] Results:
[INFO]
[ERROR] Errors:
[ERROR] ContractAgreementHandlerTest.handleMessage_validAgreement_returnMessageProcessedNotificationMessageAndLogToCH:248 ClassCast
[ERROR] ContractRequestHandlerTest.checkContractRequest_validRequest_returnOk:493 ClassCast
[ERROR] DescriptionRequestHandlerTest.handleMessage_validResourceDescriptionMsgKnownId_returnResourceDescription:176 ClassCast
[ERROR] DescriptionRequestHandlerTest.handleMessage_validSelfDescriptionMsg_returnSelfDescription:114 ClassCast
[INFO]
[ERROR] Tests run: 1341, Failures: 0, Errors: 4, Skipped: 1
[INFO]

**Comment by YogarajRethinasamy** [comment]
Am also facing the same issue

**Comment by ronjaquensel** [comment]
Hi,

could you try building the connector with `mvn clean package -DskipTests`? In productive deployment, DAT checking is always performed. As some of our tests use mocked messages with invalid DATs, maybe this is what causes the tests to fail. If this is the cause, the connector should still be able to run correctly after being built.

You can also use our pre-built Docker images. Docker-compose examples including how to set a custom *config.json* and custom properties can be found [here](https://github.com/International-Data-Spaces-Association/IDS-Deployment-Examples/tree/main/dataspace-connector/slim/docker).

**Reply by YogarajRethinasamy** [reply]
Thanks for the response, now the build is succeeded after running the build with -Dskiptests 

I have got the below message while giving my url in /api/ids/description with recipient URL/api/ids/data 

Is the below message is fine? certificate is installed correctly ?


2021-09-22T04:39:09,492 [main] INFO - Started ConnectorApplication in 23.307 seconds (JVM running for 25.563)
2021-09-22T04:39:16,717 [https-jsse-nio-6443-exec-5] INFO - Initializing Spring DispatcherServlet 'dispatcherServlet'
2021-09-22T04:39:16,718 [https-jsse-nio-6443-exec-5] INFO - Initializing Servlet 'dispatcherServlet'
2021-09-22T04:39:16,720 [https-jsse-nio-6443-exec-5] INFO - Completed initialization in 2 ms
2021-09-22T04:39:24,069 [https-jsse-nio-6443-exec-1] INFO - Init duration for springdoc-openapi is: 2631 ms
2021-09-22T04:39:34,893 [https-jsse-nio-6443-exec-4] INFO - ConnectorUUID: 7F:17:D3:32:AC:4E:BE:A8:09:72:FF:87:F6:A5:21:23:C6:AC:E4:85:keyid:CB:8C:C7:B6:85:79:A8:23:A6:CB:15:AB:17:50:2F:E6:65:43:5D:E8
2021-09-22T04:39:34,894 [https-jsse-nio-6443-exec-4] INFO - Retrieving Dynamic Attribute Token...
2021-09-22T04:39:35,265 [https-jsse-nio-6443-exec-4] INFO - Sending request to DAPS: https://daps.aisec.fraunhofer.de/v2/token
2021-09-22T04:39:35,632 [https-jsse-nio-6443-exec-4] INFO - URL is valid: null

**Reply by ghost** [reply]
We were able to determine the fingerprint of the connector by using the certificate included Connector-SKI and DAPS-AKI ("ConnectorUUID: ..."), therefore the certificate was read successfully.

How does the log continue, what does the receiver of the message log?
